third_party/move/move-prover/boogie-backend/src/options.rs

```diff
@@ -365,4 +372,32 @@ impl BoogieOptions {
         }
         Ok(())
     }
+    fn check_version_is_leq(tool: &str, given: &str, expected: &str) -> anyhow::Result<()> {
```

Comment:
Can we avoid this code duplication? This seems to do 95% the same as `check_version_is_greater`. Also, as you are as with this, rename `_greater` into `_geq` for consistency with the new function.

Reply:
I have created a separate function for string compatibility and renamed functions for consistency. The functions `check_version_is_geq` and `check_version_is_leq` are still pretty similar. I am not sure if I should further simplify.

Reply:
@aalok-t Maybe you could try to use macro rules to avoid duplication further?

Reply:
@aalok-t Or does it make sense to call `check_version_is_geq` two times in sequence with different arguments: 
Self::check_version_is_geq("z3", &version, MIN_Z3_VERSION)?;
Self::check_version_is_geq("z3", MAX_Z3_VERSION, &version,)?;

Reply:
One cannot directly do that as the error messages are slightly different. I have effectively done that in my update and will share it soon.